Tests/ForageSDKTests/CollectorTests.swift

```diff
@@ -296,6 +300,207 @@ class VaultCollectorTests: XCTestCase {
+
+        let token = "123456,789012,345678"
+        let resultToken = try forageWrapper.getPaymentMethodToken(paymentMethodToken: token)
+        XCTAssertEqual(resultToken, "345678")
+    }
+    
+    func testForageWrapper_sendData_PaymentMethodTokenError() {
+        let textElement = UITextField()
+        let config = ForageVaultConfig(environment: .sandbox)
+        let logger = MockLogger()
+        let session = URLSessionMock()
+        let forageWrapper = ForageVaultWrapper(textElement: textElement, forageVaultConfig: config, logger: logger, session: session)
+        
+        let expectation = self.expectation(description: "Completion handler called")
+        
+        forageWrapper.sendData(path: "/test/path", vaultAction: .balanceCheck, extraData: ["card_number_token": "1,2"]) { (result: MockDecodableModel?, error: ForageError?) in
+            XCTAssertNil(result)
+            XCTAssertEqual(error?.code, "unknown_server_error")
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 1)
+        
+        XCTAssertEqual(logger.lastCriticalMessage, "Failed to send data to Rosetta proxy. Rosetta token not found on card")
+        XCTAssertNil(session.lastRequest)
+    }
+    
+    func testForageWrapper_sendData_VaultProxyCallSuccess() {
+        let textElement = UITextField()
+        let config = ForageVaultConfig(environment: .sandbox)
+        let logger = MockLogger()
+        let session = URLSessionMock()
+        let forageWrapper = ForageVaultWrapper(textElement: textElement, forageVaultConfig: config, logger: logger, session: session)
+        
+        forageWrapper.customHeaders = [
+            "Session-Token": "test session token",
+            "Merchant-Account": "mid/test-merchant-id"
+        ]
+        textElement.text = "1234"
+
+        let expectation = self.expectation(description: "Completion handler called")
+        
+        forageWrapper.sendData(path: "/test/path", vaultAction: .balanceCheck, extraData: ["card_number_token": "1,2,3"]) { (result: MockDecodableModel?, error: ForageError?) in
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 1, handler: nil)
+        
+        let url = URL(string: "https://vault.sandbox.joinforage.app/proxy/test/path")!
+        let expectedBody = ["card_number_token": "3", "pin": "1234"]
+        let actualBody = try! JSONSerialization.jsonObject(with: (session.lastRequest?.httpBody)!) as! [String: String]
+        
+        XCTAssertEqual(session.lastRequest?.url, url)
+        XCTAssertEqual(session.lastRequest?.httpMethod, "POST")
+        XCTAssertEqual(actualBody, expectedBody)
+        XCTAssertEqual(session.lastRequest?.allHTTPHeaderFields, [
+            "Content-Type": "application/json",
+            "Authorization": "test session token",
+            "Merchant-Account": "mid/test-merchant-id"
+        ])
```

Comment:
if it's low lift, can we add an assertion to check that `Session-Token` is omitted from the Rosetta headers here?

Reply:
[f784b96](https://github.com/teamforage/forage-ios-sdk/pull/169/commits/f784b9650877e4f2b7b38fca8922994ac7d447a8)